pkg/collector/corechecks/servicediscovery/module/testutil/fake_server/fake_server.go

```diff
@@ -0,0 +1,31 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```

Comment:
why can't we reuse our other servers?

Reply:
The main difference (mentioned in the comment below) is that this one accepts any command line arguments. 
I guess I could change one of the other servers to ignore any command line arguments if `argv[0]` is not as expected, so that it can be used with symbolic links with different names?

Reply:
no, that make sense actually

I skipped the package comment so I didn't see the intention you had here